### PR TITLE
Add support for code coverage

### PR DESCRIPTION
**Pull request by ktf**
This bumps the version of lcov and adds the support code to integrate
PR checking with codecov.

### CODE REVIEW

**Comment by ktf** [comment]
Still requires a proper default to be enabled and and extra environment variable in the pr checker.

**Comment by ktf** [comment]
Ok, this should be it. In the end I prefer that the recipe does not know anything about codecov, and the upload happens as a side effect. This way we do not need to expose the codecov token to alibuild recipes, which is Good™.

File: o2.sh [hunk 1]
```diff
@@ -33,8 +33,23 @@ incremental_recipe: |
   fi
   if [[ $ALIBUILD_O2_TESTS ]]; then
     export O2_ROOT=$INSTALLROOT
+    # Clean up old coverage data
+    find . -name "*.gcov" -delete
```
Comment (dberzano) [inline o2.sh]: `find . -name "*.gcov" -a -name "*.gcda" -delete` (on a single line)

File: o2.sh [hunk 2]
```diff
@@ -33,8 +33,23 @@ incremental_recipe: |
...
+  # by the calling driver to codecov.io or similar service
+  if [[ $CMAKE_BUILD_TYPE == COVERAGE ]]; then
+    rm -rf coverage.info
+    lcov --base-directory $SOURCEDIR --directory . --capture --output-file coverage.info # capture coverage info
+    lcov --remove coverage.info '*/usr/*' --output-file coverage.info # filter out system
+    lcov --remove coverage.info '*/boost/*' --output-file coverage.info # filter out boost 
+    lcov --remove coverage.info '*/ROOT/*' --output-file coverage.info # filter out ROOT
+    lcov --remove coverage.info '*/FairRoot/*' --output-file coverage.info # filter out FairRoot
+    lcov --remove coverage.info '*/G__*Dict*' --output-file coverage.info # filter out FairRoot
+    lcov --list coverage.info #debug info
```
Comment (dberzano) [inline o2.sh]: `# debug info` (with a space after the pound sign)

File: o2.sh [hunk 3]
```diff
@@ -140,5 +155,21 @@ mkdir -p $INSTALLROOT/etc/modulefiles && rsync -a --delete etc/modulefiles/ $INS
 
 if [[ $ALIBUILD_O2_TESTS ]]; then
   export O2_ROOT=$INSTALLROOT
+  # Clean up old coverage data
+  find . -name "*.gcov" -delete
```
Comment (dberzano) [inline o2.sh]: Same thing as above (one line possible)

File: o2.sh [hunk 4]
```diff
@@ -140,5 +155,21 @@ mkdir -p $INSTALLROOT/etc/modulefiles && rsync -a --delete etc/modulefiles/ $INS
...
+# by the calling driver to codecov.io or similar service
+if [[ $CMAKE_BUILD_TYPE == COVERAGE ]]; then
+  rm -rf coverage.info
+  lcov --base-directory $SOURCEDIR --directory . --capture --output-file coverage.info # capture coverage info
+  lcov --remove coverage.info '*/usr/*' --output-file coverage.info # filter out system
+  lcov --remove coverage.info '*/boost/*' --output-file coverage.info # filter out boost 
+  lcov --remove coverage.info '*/ROOT/*' --output-file coverage.info # filter out ROOT
+  lcov --remove coverage.info '*/FairRoot/*' --output-file coverage.info # filter out FairRoot
+  lcov --remove coverage.info '*/G__*Dict*' --output-file coverage.info # filter out FairRoot
+  lcov --list coverage.info #debug info
```
Comment (dberzano) [inline o2.sh]: Ditto

**Pull request merged**